src/main/groovy/pluggable/scm/bitbucket/BitbucketSCMProvider.groovy

```diff
@@ -0,0 +1,241 @@
+            url.append("/")
+          }
+          if((this.scmProtocol == BitbucketSCMProtocol.HTTP) ||
+             (this.scmProtocol == BitbucketSCMProtocol.HTTPS)
+          ){
+            url.append("scm/");
+          }
+
+          return url;
+      }
+
+  /**
+  * Creates relevant repositories defined by your cartridge in your chosen SCM provider
+  * @param workspace Workspace of the cartridge loader job
+  * @param repoNamespace Location in your SCM provider where your repositories will be created
+  * @parma overwriteRepos
+  **/
+  public void createScmRepos(String workspace, String repoNamespace, String codeReviewEnabled, String overwriteRepos) {
+    URL bitbucketUrl = new URL(this.bitbucketProtocol.toString(), this.bitbucketEndpoint, this.bitbucketPort, this.bitbucketEndpointContext);
+
+    BitbucketRequestUtil.isProjectAvailable(bitbucketUrl, this.bitbucketUsername, this.bitbucketPassword, repoNamespace);
+
+    ExecuteShellCommand com = new ExecuteShellCommand()
+
+    String cartHome = "/cartridge"
+    String urlsFile = workspace + cartHome + "/src/urls.txt"
+    EnvVarProperty envVarProperty = EnvVarProperty.getInstance();
+
+    // Create repositories
+    String command1 = "cat " + urlsFile
+    List<String> repoList = new ArrayList<String>();
+    repoList = (com.executeCommand(command1).split("\\r?\\n"));
+
+    for(String repo: repoList) {
+        String repoName = repo.substring(repo.lastIndexOf("/") + 1, repo.indexOf(".git"));
+        String target_repo_name = repoNamespace + "/" + repoName
+        int repo_exists=0;
+
+        List<String> bitbucketRepoList = BitbucketRequestUtil.getProjectRepositorys(bitbucketUrl, this.bitbucketUsername, this.bitbucketPassword, repoNamespace);
+        for(String bitbucketRepo: bitbucketRepoList) {
+          if(bitbucketRepo.trim().contains(repoName)) {
+             Logger.info("Found: " + target_repo_name);
+             repo_exists=1
+             break
+          }
+        }
+        // If not, create it
+        if (repo_exists == 0) {
+          BitbucketRequestUtil.createRepository(bitbucketUrl, this.bitbucketUsername, this.bitbucketPassword, repoNamespace, repoName);
+        } else{
+          Logger.info("Repository already exists, skipping create: : " + target_repo_name);
+        }
+
+        // Populate repository
+        String tempDir = workspace + "/tmp"
+
+        def gitSsh = new File (tempDir + '/git_ssh.sh')
+        def tempScript = new File(tempDir + '/shell_script.sh')
+
+        gitSsh << "#!/bin/sh\n"
+        gitSsh << "exec ssh -i " + envVarProperty.getSshPrivateKeyPath() + " -o StrictHostKeyChecking=no \"\$@\""
+
+        tempScript << "export GIT_SSH=\""+ tempDir + "/git_ssh.sh\"\n"
+        tempScript << "git clone " + this.getScmUrl(this.bitbucketUsername, this.bitbucketPassword) + repoNamespace + "/" + repoName + ".git " + tempDir + "/" + repoName + "\n"
+        def gitDir = "--git-dir=" + tempDir + "/" + repoName + "/.git"
```

Comment:
BitBucket password should be ASCII encoded as it's a url. What if someone has an '@' in their password?